sync-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -377,12 +381,18 @@ private int processOneChangesBatch(ChangesResultWrapper changeFeeds)
         return changesProcessed;
     }
 
-    private String getReplicationId() {
-        if(filter == null) {
-            return this.sourceDb.getIdentifier() ;
-        } else {
-            return this.sourceDb.getIdentifier() + "?" + filter.toQueryString();
+    public String getReplicationId() {
+        HashMap<String, String> dict = new HashMap<String, String>();
+        dict.put("source", this.sourceDb.getIdentifier());
+        dict.put("target", this.targetDb.getDbCore().getPublicIdentifier());
+        if(filter != null) {
+            dict.put("filter", this.filter.toQueryString());
         }
+        // get raw SHA-1 of dictionary
+        byte[] sha1Bytes = Misc.getSha1(new ByteArrayInputStream(JSONUtils.serializeAsBytes(dict)));
+        // return SHA-1 as a hex string
+        byte[] sha1Hex = new Hex().encode(sha1Bytes);
+        return new String(sha1Hex);
```

Comment:
It's in the same spirit but not an exact copy. We use "source" and "target" rather than "localUUID" and "remoteURL".

The iOS code is this:
```
    NSMutableDictionary* spec =
        $mdict({ @"localUUID", _db.privateUUID }, { @"remoteURL", _remote.absoluteString },
               { @"push", @(self.isPush) }, { @"filter", _filterName },
               { @"filterParams", _filterParameters });
    return TDHexSHA1Digest([TDCanonicalJSON canonicalData:spec]);
```